python/pyspark/sql/tests/test_readwriter.py

```diff
@@ -53,16 +53,23 @@ def test_save_and_load(self):
         )
         self.assertEqual(sorted(df.collect()), sorted(actual.collect()))
 
-        defaultDataSourceName = self.spark.conf.get(
-            "spark.sql.sources.default", "org.apache.spark.sql.parquet"
-        )
-        self.spark.sql("SET spark.sql.sources.default=org.apache.spark.sql.json")
-        actual = self.spark.read.load(path=tmpPath)
-        self.assertEqual(sorted(df.collect()), sorted(actual.collect()))
-        self.spark.sql("SET spark.sql.sources.default=" + defaultDataSourceName)
+        try:
```

Comment:
Hm, let's avoid such changes like this because of inherited test cases. I think `test_readwriter` itself should make sense itself alone (with regular PySpark alone).

To do this, one thing we can do is to split the test separately. E.g., https://github.com/apache/spark/commit/56f8dccd6532aaebf9f2838be6673d5c75c6df79

Reply:
got it, will update.

Reply:
I found that is seems not easy to split the test, since it depends on the previous lines and I almost have to copy all the tests. So I just update the related tickets here.